File: src/main/java/io/realworld/api/UserAndAuthenticationApi.java
package io.realworld.api;

import io.realworld.model.Login200Response;

import io.realworld.model.request.CreateUser;
import io.realworld.model.request.Login;
import io.realworld.model.request.UpdateUser;
import io.vertx.core.Future;
import io.vertx.ext.web.RoutingContext;

public interface UserAndAuthenticationApi  {
    Future<ApiResponse<Login200Response>> createUser(CreateUser body);
    Future<ApiResponse<Login200Response>> getCurrentUser(RoutingContext rc);
    Future<ApiResponse<Login200Response>> login(Login body);
    Future<ApiResponse<Login200Response>> updateCurrentUser(UpdateUser body);
}

File: src/main/kotlin/io/realworld/api/UserAndAuthenticationApi.kt
package io.realworld.api

import io.realworld.model.request.CreateUserRequest
import io.realworld.model.request.LoginRequest
import io.realworld.model.request.UpdateCurrentUserRequest
import io.realworld.model.response.ApiResponse
import io.realworld.model.response.Login200Response
import io.vertx.core.Future
import io.vertx.ext.web.RoutingContext

interface UserAndAuthenticationApi {
  fun createUser(body: CreateUserRequest): Future<ApiResponse<Login200Response>>
  fun getCurrentUser(rc: RoutingContext): Future<ApiResponse<Login200Response>>
  fun login(body: LoginRequest): Future<ApiResponse<Login200Response>>
  fun updateCurrentUser(body: UpdateCurrentUserRequest): Future<ApiResponse<Login200Response>>
}
